rstcheck/__init__.py

```diff
@@ -517,17 +523,41 @@ def load_configuration_from_file(directory: str, args: argparse.Namespace) -> ar
 
 def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
     config_path = find_config(directory_or_file, debug=debug)
-    if not config_path:
+    if not config_path:  # pylint: disable=no-else-return
         return {}
+    else:
+        config_file = os.path.basename(config_path)
 
-    parser = configparser.ConfigParser()
-    parser.read(config_path)
-    try:
-        options = dict(parser.items("rstcheck"))
-    except configparser.NoSectionError:
-        return {}
+    if config_file == "pyproject.toml" and TOMLI_INSTALLED:  # pylint: disable=no-else-return
+        return _get_pyproject_options(config_path)
     else:
-        return options
+        parser = configparser.ConfigParser()
+        parser.read(config_path)
+        try:
+            options = dict(parser.items("rstcheck"))
+        except configparser.NoSectionError:
+            return {}
+        else:
+            return options
+
+
+def _get_pyproject_options(config_path: str) -> typing.Dict[str, str]:
+    with open(config_path, "rb") as conf_file:
+        config = tomli.load(conf_file)
+        options: typing.Dict[str, str] = config.get("tool", {}).get("rstcheck", {})
+
+    # tomli returns a list of strings and ConfigParser returns a comma
+    # separated string.  This makes the options from pyproject.toml consistent
+    # with the options read from other configuration files.  The try block
+    # accounts for pyproject files without a rstcheck section.
+    for _ignore in ["ignore_directives", "ignore_roles", "ignore_messages", "ignore_language"]:
```

Comment:
I didn't include ```report``` in this originally because that option can only take one value.  The ignore options can take more than one and, thus, require a list in pyproject.toml.  It's only these list options that cause problems and need to be converted to a comma separated string.  I changed the conditional to allow any of the options to be included in pyproject.toml as a string or a list of strings.  You'll see I also changed the ```ignore_language``` option in the example pyproject.toml from ```["cpp"]``` to ```"cpp"``` to cover this case.

I also wonder if ```recursive``` and/or ```debug``` should be added as well to completely cover all rstcheck options?  Again, this would be scope creep for this PR, so I can raise an issue if you agree to the idea and want to tackle it later.

Reply:
`report` is a config value that is currently supported by the other config files and I think that is good. Therefore I would like pyproject.toml to do the same.

The config file is loaded for each file in their respective directory. This way you can set the report level in a nested config file without affecting files up the file tree. IMHO this is a valid reason. A corner case but valid.

`recursive` makes only sense if you give a directory. Therefore it IMO makes no sense to put into a config file because you specify no files/directories here.

`debug` is a special option you use actively and therefore it IMO makes also no sense to be put into a config file.

Further discussions please in a separate issue.

Reply:
I think I wasn't clear in my earlier comment.  I included ```report``` support in pyproject.toml from the beginning.  But, because it can only be one value (info, warning, error, severe, none), there is no need for it to be a list in pyproject.toml.  The ignore options can take multiple values which requires a list in pyproject.toml.  The for loop, as I originally had it, was only converting the options that must be lists for multiple values in pyproject.toml to comma-separated strings.

This is what is returned when reading pyproject.toml:
```{'ignore_directives': ['foobar', 'my-directive'], 'ignore_roles': ['some-custom-thing'], 'ignore_messages': ['(Document or section may not begin with a transition\\.$)'], 'ignore_language': ['cpp'], 'report': 'warning'}```

This is the dict created from dict(configparser.items("rstcheck")): 
```{'ignore_directives': '\nfoobar,\nmy-directive,', 'ignore_roles': 'some-custom-thing', 'ignore_messages': '(Document or section may not begin with a transition\\.$)', 'ignore_language': 'cpp', 'report': 'warning'}```

Since the remainder of the rstcheck functions are expecting the configparser format, the for loop is making the pyproject.toml output compatible (i.e., ```['foobar', 'my-directive']``` becomes ```'foobar,my-directive,'```).

When adding ```report``` to the list of options to convert, it becomes necessary to put ```report = ["warning"]``` in pyproject.toml otherwise the for loop changes it to ```"w,a,r,n,i,n,g,"```.  By having the conditional check if the option is a list and only convert it if so, it allows the user the option of ```report = ["warning"]``` **or** ```report = "warning"``` in pyproject.toml.  This also means **any** of the config options can be either a list or a string which I think is less constraining on the user.  It also more closely follows the format of .rstcheck.cfg in that single value options are simply  ```report=warning```.

See also my comment below.